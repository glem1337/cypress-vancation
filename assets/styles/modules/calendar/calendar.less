@calendar-item-width: 50px;
@calendar-item-width-md: 72px;
@calendar-item-width-lg: 84px;
@calendar-item-height: 40px;

.calendar {
  display: block;
  width: 100%;
  max-width: 1436px;
  margin-right: auto;
  margin-left: auto;
}

.calendar__head {
  .flex-center-x();
  position: relative;
  padding: 72px 0 16px;

  @media (min-width: @screen-md) {
    padding: 24px 0;
  }

  @media (min-width: @screen-lg) {
    margin-left: 260px;
  }
}

.calendar__head-today-btn {
  position: absolute;
  top: 16px;
  left: 16px;

  @media (min-width: @screen-md) {
    top: 50%;
    left: 32px;
    transform: translateY(-50%);
  }

  @media (min-width: @screen-lg) {
    left: 0;
  }
}

.calendar__head-switch {
  position: absolute;
  top: 16px;
  right: 16px;

  @media (min-width: @screen-md) {
    top: 50%;
    right: 32px;
    transform: translateY(-50%);
  }

  @media (min-width: @screen-lg) {
    right: 24px;
  }
}

.calendar__wrap {
  display: flex;
  position: relative;
  width: 100%;
  overflow: hidden;
}

.calendar__side {
  position: sticky;
  z-index: @calendar-side-z-index;
  flex-shrink: 0;
  width: 196px;
  transition: margin-left 0.3s;
  background-color: white;
  box-shadow: @shadow2;

  @media (min-width: @screen-md) {
    width: 260px;
  }
}

.calendar__side--hidden {
  margin-left: -180px;
}

.calendar__side-control {
  position: absolute;
  top: calc(50% + 20px);
  right: -16px;
  transform: translateY(-50%);

  @media (min-width: @screen-md) {
    display: none;
  }
}

.calendar__side-head {
  .flex-center-y();
  height: @calendar-item-height;
  padding-left: 8px;
  padding-right: 8px;
  font-family: @alt-font-family;
  font-weight: 700;
  background-color: @gray200;
  border-bottom: 1px solid @gray300;
  color: @black;

  @media (min-width: @screen-md) {
    font-size: 1.6rem;
  }
}

.calendar__side-item {
  .flex-center-y();
  height: @calendar-item-width-lg;
  padding: 8px;
  border-bottom: 1px solid @gray300;
}

.calendar__side-item-img {
  flex: none;
  width: 68px;
  height: 68px;
  margin-right: 8px;
  border-radius: @main-border-radius;
  overflow: hidden;

  img {
    display: block;
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
}

.calendar__side-item-descr {
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: flex-start;
}

.calendar__side-item-link {
  display: block;
  display: -webkit-box;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 2;
  margin-bottom: 8px;
  font-size: 1.2rem;
  font-weight: 600;

  @media (min-width: @screen-md) {
    -webkit-line-clamp: 1;
    font-size: 1.4rem;
  }
}

.calendar__body {
  flex: 1;
  overflow-x: auto;
}

.calendar__body-head {
  display: flex;
}

.calendar__body-head-item {
  .flex-center();
  flex: none;
  width: @calendar-item-width;
  height: @calendar-item-height;
  border-top: 1px solid @gray300;
  border-bottom: 1px solid @gray300;
  color: @gray700;
  text-transform: uppercase;
  font-size: 1rem;

  @media (min-width: @screen-md) {
    width: @calendar-item-width-md;
  }

  @media (min-width: @screen-lg) {
    width: @calendar-item-width-lg;
  }
}

.calendar__body-row {
  display: flex;
}

.calendar__body-cell {
  display: flex;
  position: relative;
  flex: none;
  align-items: flex-end;
  justify-content: flex-end;
  width: @calendar-item-width;
  height: @calendar-item-width-lg;
  padding: 8px;
  box-shadow: inset -1px -1px @gray400;
  color: @black;

  @media (min-width: @screen-md) {
    width: @calendar-item-width-md;
  }

  @media (min-width: @screen-lg) {
    width: @calendar-item-width-lg;
  }

  &::before {
    content: '';
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: @calendar-cell-z-index;
    background: @calendar-cell-bg;
  }
}

.calendar__body-cell--active::before {
  background: white;
}

.calendar__body-cell--passed {
  box-shadow: inset -1px -1px rgba(@gray400, 0.5);

  &::before {
    opacity: 0.5;
  }

  .calendar__body-cell-price {
    opacity: 0.5;
  }
}

.calendar__body-cell--last-passed::after {
  content: '';
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  border-right: 1px solid @black;
}

.calendar__body-cell-booking {
  .flex-center-y();
  position: absolute;
  top: 50%;
  z-index: @calendar-booking-z-index;
  transform: translateY(-50%);
  transition: box-shadow 0.3s;
  height: 24px;
  padding-left: 10px;
  padding-right: 8px;
  background-color: @blue1000;
  color: white;
  border-radius: @small-border-radius;
  overflow: hidden;
  cursor: pointer;

  &::after {
    content: '';
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    width: 4px;
    background-color: @blue-accent;
  }
}

.calendar__body-cell-booking--is-active {
  box-shadow: @shadow3;
}

.calendar__body-cell-booking--before-range {
  padding-left: 8px;
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;

  &::after {
    display: none;
  }
}

.calendar__body-cell-booking--after-range {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}

.calendar__body-cell-booking-info {
  flex: 1;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-size: 1.2rem;
  user-select: none;
}

.calendar__body-cell-price {
  font-size: 1.2rem;
}

.calendar__booking-popover {
  display: flex;
  flex-direction: column;
  max-width: 100%;

  @media (min-width: @screen-md) {
    flex-direction: row;
    align-items: center;
    flex: 1;
  }

  @media (min-width: @screen-lg) {
    flex-direction: column;
    flex: none;
  }
}

.calendar__booking-popover-close {
  position: absolute;
  top: 8px;
  right: 12px;
}
