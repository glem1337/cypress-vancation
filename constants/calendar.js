import moment from 'moment';
import isPresent from 'utils/isPresent';

export const CALENDAR_BOUNDARIES = {
  MIN: moment().subtract(12, 'month').startOf('month'),
  MAX: moment().add(37, 'month').startOf('month'),
};

export const FETCH_CAMPER_CALENDAR_BOUNDARIES = (date) => {
  const baseDate = isPresent(date)
    ? moment(date)
    : moment();

  return {
    startDate: baseDate.clone().subtract(1, 'M').startOf('month').format('YYYY-MM-DD'),
    endDate: baseDate.clone().add(1, 'M').endOf('month').format('YYYY-MM-DD'),
  };
};

export const AVAILABILITY_MODE = {
  AVAILABLE: 'available',
  BLOCKED: 'blocked',
};

export const CALENDAR_AVAILABILITY_DEFAULT_VALUE = {
  value: 6,
    label: {
      id: 'shared.countMonth',
      values: { count: 6 },
    },
};

export const CALENDAR_AVAILABILITY_OPTIONS = [
  {
    value: 3,
    label: {
      id: 'shared.countMonth',
      values: { count: 3 },
    },
  },
  {
    value: 6,
    label: {
      id: 'shared.countMonth',
      values: { count: 6 },
    },
  },
  {
    value: 12,
    label: {
      id: 'shared.countMonth',
      values: { count: 12 },
    },
  },
  {
    value: 24,
    label: {
      id: 'shared.countMonth',
      values: { count: 24 },
    },
  },
  {
    value: 36,
    label: {
      id: 'shared.countMonth',
      values: { count: 36 },
    },
  },
];

export const CALENDAR_PREPARATION_TIME_DEFAULT_VALUE = {
  value: 0.1,
  label: {
    id: 'shared.countMonth',
    values: { count: 0 },
  },
};

export const CALENDAR_PREPARATION_TIME_OPTIONS = [
  {
    value: 0.1,
    label: {
      id: 'shared.countDays',
      values: { count: 0 },
    },
  },
  {
    value: 1,
    label: {
      id: 'shared.countDays',
      values: { count: 1 },
    },
  },
  {
    value: 2,
    label: {
      id: 'shared.countDays',
      values: { count: 2 },
    },
  },
  {
    value: 3,
    label: {
      id: 'shared.countDays',
      values: { count: 3 },
    },
  },
  {
    value: 4,
    label: {
      id: 'shared.countDays',
      values: { count: 4 },
    },
  },
  {
    value: 5,
    label: {
      id: 'shared.countDays',
      values: { count: 5 },
    },
  },
];

export const CALENDAR_EVENT_TYPE = {
  BLOCKED: 'blocked',
  BOOKED: 'booked',
  EXTERNAL: 'external',
};
