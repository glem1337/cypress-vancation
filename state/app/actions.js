import {
  SET_PAGES,
  SET_CURRENT_LOCATION,
  SET_CURRENT_COORDINATES,
  SET_OPEN_GRAPH_DATA,
} from './type';

export const setPages = ({ currentPage, prevPage, isServer }) => ({
  type: SET_PAGES,
  isServer: isServer || false,
  currentPage,
  prevPage,
});

export const setCurrentCoordinates = ({ longitude, latitude }) => ({
  type: SET_CURRENT_COORDINATES,
  longitude,
  latitude,
});

export const setCurrentLocation = (location) => ({
  type: SET_CURRENT_LOCATION,
  location,
});

export const setOpenGraphData = (data) => ({
  type: SET_OPEN_GRAPH_DATA,
  data,
});
