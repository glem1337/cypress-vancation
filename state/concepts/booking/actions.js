import makeFormSubmitAction from 'utils/form/makeFormSubmitAction';

import {
  FETCH_CAMPER_BOOKING_INQUIRY,
  CREATE_CAMPER_BOOKING_INQUIRY,
  SET_CHAT_SECTION_VISIBILITY,
  SET_DETAILS_SECTION_VISIBILITY,
  FETCH_CAMPER_INQUIRIES,
  SET_CAMPER_INQUIRIES_IDS,
  SET_CAMPER_INQUIRIES_PAGE,
  SET_CAMPER_INQUIRIES_TOTAL,
  SET_FILTERS_VISIBILITY,
  SET_ACTIVE_BOOKING_ID,
  FETCH_BOOKING_CHAT_MESSAGES,
  SET_BOOKING_CHAT_MESSAGES_IDS,
  SET_BOOKING_CHAT_MESSAGES_PAGE,
  SET_BOOKING_CHAT_MESSAGES_TOTAL,
} from './types';

export const createCamperBookingInquiry = makeFormSubmitAction(
  CREATE_CAMPER_BOOKING_INQUIRY,
);

export const fetchCamperBookingInquiry = (id) => ({
  type: FETCH_CAMPER_BOOKING_INQUIRY,
  id,
});

export const setChatSectionVisibility = (isVisible) => ({
  type: SET_CHAT_SECTION_VISIBILITY,
  isVisible,
});

export const setDetailsSectionVisibility = (isVisible) => ({
  type: SET_DETAILS_SECTION_VISIBILITY,
  isVisible,
});

export const fetchCamperInquiries = ({
  page = 1,
  perPage = 20,
  userType,
} = {}) => ({
  type: FETCH_CAMPER_INQUIRIES,
  page,
  perPage,
  userType,
});

export const setCamperInquiriesIds = (ids) => ({
  type: SET_CAMPER_INQUIRIES_IDS,
  ids,
});

export const setCamperInquiriesPage = (page) => ({
  type: SET_CAMPER_INQUIRIES_PAGE,
  page,
});

export const setCamperInquiriesTotal = (total) => ({
  type: SET_CAMPER_INQUIRIES_TOTAL,
  total,
});

export const setFiltersVisibility = (isVisible) => ({
  type: SET_FILTERS_VISIBILITY,
  isVisible,
});

export const setActiveBookingId = (id) => ({
  type: SET_ACTIVE_BOOKING_ID,
  id,
});

export const fetchBookingChatMessages = ({
  id = null,
  page = 1,
  perPage = 20,
} = {}) => ({
  type: FETCH_BOOKING_CHAT_MESSAGES,
  id,
  page,
  perPage,
});

export const setBookingChatMessagesIds = (ids) => ({
  type: SET_BOOKING_CHAT_MESSAGES_IDS,
  ids,
});

export const setBookingChatMessagesPage = (page) => ({
  type: SET_BOOKING_CHAT_MESSAGES_PAGE,
  page,
});

export const setBookingChatMessagesTotal = (total) => ({
  type: SET_BOOKING_CHAT_MESSAGES_TOTAL,
  total,
});
