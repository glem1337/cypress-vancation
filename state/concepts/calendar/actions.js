import {
  SET_CURRENT_DATE,
  SET_SETTINGS_VISIBILITY,
  SET_SELECTED_SLOTS,
  CLEAR_SELECTED_SLOTS,
  SET_AVAILABILITY_VISIBILITY,
  SET_AVAILABILITY_MODE,
  CLOSE_RIGHT_SIDEBAR,
  TOGGLE_CALENDAR_FOOTER,
  FIND_EXTERNAL_CALENDAR_IDS,
  SET_EXTERNAL_CALENDAR_IDS,
  REMOVE_EXTERNAL_CALENDAR,
  SET_CUSTOM_DISCOUNT_PERIODS_IDS,
  SET_CUSTOM_NIGHT_RATES_IDS,
  SET_BLOCKED_PERIODS_IDS,
  SET_CUSTOM_MIN_NIGHT_STAY_IDS,
  SET_EXTERNAL_EVENT_IDS,
} from './types';

export const setCurrentDate = (date) => ({
  type: SET_CURRENT_DATE,
  date,
});

export const setCalendarSettingsVisibility = (isVisible) => ({
  type: SET_SETTINGS_VISIBILITY,
  isVisible,
});

export const setAvailabilityVisibility = (isVisible) => ({
  type: SET_AVAILABILITY_VISIBILITY,
  isVisible,
});

export const setAvailabilityMode = (mode) => ({
  type: SET_AVAILABILITY_MODE,
  mode,
});

export const closeRightSidebar = () => ({
  type: CLOSE_RIGHT_SIDEBAR,
});

export const setSelectedSlots = ({ start, end, slots }) => ({
  type: SET_SELECTED_SLOTS,
  start,
  end,
  slots,
});

export const clearSelectedSlots = () => ({
  type: CLEAR_SELECTED_SLOTS,
});

export const toggleCalendarFooter = () => ({
  type: TOGGLE_CALENDAR_FOOTER,
});

export const findExternalCalendarIds = (response) => ({
  type: FIND_EXTERNAL_CALENDAR_IDS,
  response,
});

export const setExternalCalendarIds = (ids) => ({
  type: SET_EXTERNAL_CALENDAR_IDS,
  ids,
});

export const removeExternalCalendar = (id) => ({
  type: REMOVE_EXTERNAL_CALENDAR,
  id,
});

export const setCustomDiscountPeriodIds = (ids) => ({
  type: SET_CUSTOM_DISCOUNT_PERIODS_IDS,
  ids,
});

export const setCustomNightRatesIds = (ids) => ({
  type: SET_CUSTOM_NIGHT_RATES_IDS,
  ids,
});

export const setCustomMinNightStayIds = (ids) => ({
  type: SET_CUSTOM_MIN_NIGHT_STAY_IDS,
  ids,
});

export const setBlockedPeriodIds = (ids) => ({
  type: SET_BLOCKED_PERIODS_IDS,
  ids,
});

export const setExternalEventsIds = (ids) => ({
  type: SET_EXTERNAL_EVENT_IDS,
  ids,
});
