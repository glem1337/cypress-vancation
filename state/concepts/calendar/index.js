import calendarReducer from 'state/concepts/calendar/reducer';

export { default as calendarOperations } from './operations';
export * as camperRentalTypes from './types';

export default calendarReducer;
