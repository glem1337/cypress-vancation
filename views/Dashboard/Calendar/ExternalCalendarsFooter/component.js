import PropTypes from 'prop-types';
import { Button } from 'antd';
import { FormattedMessage } from 'react-intl';

import ExportCalendarSection from './ExportCalendarSection';
import ImportCalendarSection from './ImportCalendarSection';

const ExternalCalendarsFooter = ({ toggleFooter, camperId }) => (
  <div className="calenadr-listing__footer">
    <div className="calendar-listing__footer-head">
      <FormattedMessage id="calendar.footer.title" />
      <Button
        className="d-lg-none align-self-start ml-16"
        type="secondary"
        icon={<i className="icon icon-cross" />}
        shape="circle"
        size="large"
        onClick={toggleFooter}
      />
    </div>
    <div className="calendar-listing__footer-main">
      <ExportCalendarSection camperId={camperId} />
      <ImportCalendarSection camperId={camperId} />
    </div>
  </div>
);

ExternalCalendarsFooter.propTypes = {
  toggleFooter: PropTypes.func.isRequired,
  camperId: PropTypes.string.isRequired,
};

export default ExternalCalendarsFooter;
