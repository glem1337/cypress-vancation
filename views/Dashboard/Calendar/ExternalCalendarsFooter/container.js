import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { toggleCalendarFooter as toggleCalendarFooterAction } from 'state/concepts/calendar/actions';

import ExternalCalendarsFooterComponent from './component';

class ExternalCalendarsFooter extends React.Component {
  static propTypes = {
    toggleCalendarFooter: PropTypes.func.isRequired,
  };

  toggleFooter = (event) => {
    const { toggleCalendarFooter } = this.props;

    event.preventDefault();

    toggleCalendarFooter();
  };

  render() {
    return (
      <ExternalCalendarsFooterComponent
        {...this.props}
        toggleFooter={this.toggleFooter}
      />
    );
  }
}

const mapDispatchToProps = {
  toggleCalendarFooter: toggleCalendarFooterAction,
};

export { ExternalCalendarsFooter as ExternalCalendarsFooterContainer };
export default connect(null, mapDispatchToProps)(ExternalCalendarsFooter);
