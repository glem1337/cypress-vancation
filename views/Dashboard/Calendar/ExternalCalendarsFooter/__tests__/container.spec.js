import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import diveTo from 'utils/testHelpers/diveToEnzyme';
import { toggleCalendarFooter } from 'state/concepts/calendar/actions';

import ExternalCalendarsFooter, {
  ExternalCalendarsFooterContainer,
} from '../container';

describe('ExternalCalendarsFooter container tests', () => {
  const store = configureStore()({});
  store.dispatch = jest.fn();

  const props = {
    store,
    toggleCalendarFooter: jest.fn(),
    camperId: 'camperId',
  };

  const wrapper = shallow(<ExternalCalendarsFooter {...props} />);
  const container = diveTo(wrapper, ExternalCalendarsFooterContainer);
  const instance = container.instance();

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot();
  });

  it('checks `toggleFooter` instance method', () => {
    const event = {
      preventDefault: jest.fn(),
    };

    instance.toggleFooter(event);

    expect(event.preventDefault).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledWith(toggleCalendarFooter());
  });
});
