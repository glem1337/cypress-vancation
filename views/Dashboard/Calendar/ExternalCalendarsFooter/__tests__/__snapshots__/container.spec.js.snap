// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalCalendarsFooter container tests matches snapshot 1`] = `
<ExternalCalendarsFooter
  camperId="camperId"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  toggleCalendarFooter={[Function]}
  toggleFooter={[Function]}
/>
`;
